website/docs/plugin/framework/functions/implementation.mdx

```diff
@@ -0,0 +1,252 @@
+}
+```
+
+### Metadata Method
+
+The [`function.Function` interface `Metadata` method](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/function#Function.Metadata) defines the function name as it would appear in Terraform configurations. Unlike resources and data sources, this name should **NOT** include the provider name as the configuration language syntax for calling functions will separately include the provider name. Refer to [naming](/terraform/plugin/best-practices/naming) for additional best practice details.
+
+In this example, the function name is set to `example`:
+
+```go
+// With the function.Function implementation
+func (f *ExampleFunction) Metadata(ctx context.Context, req function.MetadataRequest, resp *function.MetadataResponse) {
+    resp.Name = "example"
+}
+```
+
+### Definition Method
+
+The [`function.Function` interface `Definition` method](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/function#Function.Definition) defines the parameters, return, and various descriptions for documentation of the function.
+
+In this example, the function definition includes one string parameter, a string return, and descriptions for documentation:
+
+```go
+func (f *ExampleFunction) Definition(ctx context.Context, req function.DefinitionRequest, resp *function.DefinitionResponse) {
+    resp.Definition = function.Definition{
+        Summary:     "Echo a string",
+        Description: "Given a string value, returns the same value.",
+
+        Parameters: []function.Parameter{
+            function.StringParameter{
+                Description: "Value to echo",
+                Name:        "input",
+            },
+        },
+        Return: function.StringReturn{},
+    }
+}
+```
+
+#### Return
+
+The `Return` field must be defined as all functions must return a result. This influences how the [Run method](#run-method) must set the result data. Refer to the [returns](/terraform/plugin/framework/functions/returns) documentation for details about all available types and how to handle data with each type.
+
+#### Parameters
+
+There may be zero or more parameters, which are defined with the `Parameters` field. They are ordered, which influences how practitioners call the function in their configurations and how the [Run method](#run-method) must read the argument data. Refer to the [parameters](/terraform/plugin/framework/functions/parameters) documentation for details about all available types and how to handle data with each type.
+
+An optional `VariadicParameter` field enables a final variadic parameter which accepts zero, one, or more values. It may be optionally combined with `Parameters`, meaning it represents the any argument data after the final parameter. When reading argument data, a `VariadicParameter` is represented as an ordered list of the parameter type, where the list has zero or more elements to match the given arguments.
```

Comment:
Potentially obvious but should we call out that all of the values are of the same type?

```suggestion
An optional `VariadicParameter` field enables a final variadic parameter which accepts zero, one, or more values of the same type. It may be optionally combined with `Parameters`, meaning it represents the any argument data after the final parameter. When reading argument data, a `VariadicParameter` is represented as an ordered list of the parameter type, where the list has zero or more elements to match the given arguments.
```